sdk/src/ed25519_instruction.rs

```diff
@@ -0,0 +1,329 @@
+#![cfg(feature = "full")]
+
+use crate::{decode_error::DecodeError, instruction::Instruction};
+use bytemuck::{bytes_of, Pod, Zeroable};
+use ed25519_dalek::{ed25519::signature::Signature, Signer, Verifier};
+use serde_derive::{Deserialize, Serialize};
+use thiserror::Error;
+
+#[derive(Error, Debug, Serialize, Clone, PartialEq)]
+pub enum Ed25519Error {
+    #[error("ed25519 public key is not valid")]
+    InvalidPublicKey,
+    #[error("ed25519 signature is not valid")]
+    InvalidSignature,
+    #[error("offset not valid")]
+    InvalidDataOffsets,
+    #[error("instruction is incorrect size")]
+    InvalidInstructionDataSize,
+}
+
+impl<T> DecodeError<T> for Ed25519Error {
+    fn type_of() -> &'static str {
+        "Ed25519Error"
+    }
+}
+
+pub const PUBKEY_SERIALIZED_SIZE: usize = 32;
+pub const SIGNATURE_SERIALIZED_SIZE: usize = 64;
+pub const SIGNATURE_OFFSETS_SERIALIZED_SIZE: usize = 14;
+pub const NUM_SIGNATURES_SIZE: usize = 2;
+pub const DATA_START: usize = SIGNATURE_OFFSETS_SERIALIZED_SIZE + NUM_SIGNATURES_SIZE;
+
+#[derive(Default, Debug, Deserialize, Copy, Clone, Zeroable, Pod)]
```

Comment:
`Deserialize` can be removed now that bincode has been given the 👢

Reply:
Indeed it can, removed